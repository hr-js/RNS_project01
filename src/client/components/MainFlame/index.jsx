// @flow
import * as React from "react";
import styles from "./style.css";

type PropType = {
  children: React.Node
};

function MainFlame(props: PropType): React.Node {
  const { children } = props;
  const { main } = styles;
  return <main className={main}>{children}</main>;
}
export default MainFlame;
