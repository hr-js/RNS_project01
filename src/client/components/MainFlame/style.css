@charset "utf-8";
.main {
  display: flex;
  position: relative;
  width: calc(100% - 170px);
  height: 100%;
  margin-left: 170px;
  box-sizing: border-box;
  padding: 40px;
}

/* デフォルト背景 */
.standard {
  background-color: #90ee90;
}

/* 仕事 */
.work {
  background-color: #bbfeff;
}

/* 自習 */
.study {
  background-color: #bcf3cb;
}

/* 勉強会 */
.meetUp {
  background-color: #f8f189;
}

/* サークル */
.circle {
  background-color: #f4d8f5;
}

/* 退出 */
.exit {
  background-color: #00558f;
}

/* エラー */
.error {
  background-color: #ea0032;
}
