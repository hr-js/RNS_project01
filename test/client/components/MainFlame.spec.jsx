import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import MainFlame from "@components/MainFlame";

// eslint-disable-next-line flowtype/require-return-type
const TestComponent = () => <div>test component</div>;

describe("MainFlame.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer
        .create(
          <MainFlame>
            <TestComponent />
          </MainFlame>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    it("main要素をもつ", () => {
      const mainFlame = shallow(
        <MainFlame>
          <TestComponent />
        </MainFlame>
      );
      expect(mainFlame.name()).toBe("main");
    });

    it("cssクラス: mainをもつ", () => {
      const mainFlame = shallow(
        <MainFlame>
          <TestComponent />
        </MainFlame>
      );
      expect(
        mainFlame
          .find("main")
          .at(0)
          .hasClass("main")
      ).toBeTruthy();
    });

    it("指定したコンポーネントを子コンポーネントとしてもつ", () => {
      const mainFlame = mount(
        <MainFlame>
          <TestComponent />
        </MainFlame>
      );
      expect(mainFlame.children().text()).toBe("test component");
    });
  });
});
