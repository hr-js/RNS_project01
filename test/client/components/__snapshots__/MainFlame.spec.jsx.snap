// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainFlame.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<main
  className="main"
>
  <div>
    test component
  </div>
</main>
`;
